ppoprf/src/lib.rs

```diff
@@ -15,8 +15,22 @@
 pub mod ggm;
 pub mod ppoprf;
 
+use derive_more::{Error, Display};
+
+#[derive(Debug, Error, Display, PartialEq)]
+pub enum PPRFError {
+    #[display(fmt = "Tag index is out of bounds, indicated index {} is above length of {}", index, tag_size)]
+    BadTagIndex { index: usize, tag_size: usize },
+    #[display(fmt = "No prefix found")]
+    NoPrefixFound,
+    #[display(fmt = "Tag already punctured")]
+    AlreadyPunctured,
+    #[display(fmt = "Input length ({}) does not match input param ({})", actual, expected)]
+    BadInputLength { actual: usize, expected: usize }
```

Comment:
`BadInputValue` may be possible too, if the point is off-curve? I suppose that can be added later.